internal/monitors/kubernetes/cluster/metrics/cache.go

```diff
@@ -210,16 +197,45 @@ func (dc *DatapointCache) handleAddService(svc *v1.Service) {
 func (dc *DatapointCache) handleDeleteService(svcUID types.UID) {
 	pods := dc.serviceCache.DeleteByKey(svcUID)
 	for _, podUID := range pods {
-		cachedPod := dc.podCache.GetCachedPod(podUID)
-		if cachedPod != nil {
-			dimProps := dimPropsForPod(cachedPod, dc.serviceCache)
-			if dimProps != nil {
-				dc.addDimPropsToCache(podUID, dimProps)
-			}
+		dc.refreshPodDimProps(podUID)
+	}
+}
+
+// handleAddReplicaSet adds a replicaset to the internal cache and
+// returns the datapoints and dimProps for the replicaset.
+func (dc *DatapointCache) handleAddReplicaSet(rs *v1beta1.ReplicaSet) ([]*datapoint.Datapoint, *atypes.DimProperties) {
+	if !dc.replicaSetCache.IsCached(rs) {
+		dc.replicaSetCache.AddReplicaSet(rs)
+		podUIDs := dc.podCache.GetPodsInNamespace(rs.Namespace)
+		dc.refreshPodsDimProps(podUIDs)
+	}
+	dps := datapointsForReplicaSet(rs)
+	dimProps := dimPropsForReplicaSet(rs)
+	return dps, dimProps
+}
+
+// refreshPodDimProps calls dimPropsForPod to update a pod's
+// properties in the dimProps cache.
+// We need to do this occasionally because of the dynamic nature
+// of a pods properties
+func (dc *DatapointCache) refreshPodDimProps(uid types.UID) {
+	cachedPod := dc.podCache.GetCachedPod(uid)
+	if cachedPod != nil {
+		dimProps := dimPropsForPod(cachedPod, dc.serviceCache, dc.replicaSetCache)
+		if dimProps != nil {
+			dc.addDimPropsToCache(uid, dimProps)
 		}
 	}
 }
 
+// refreshPodsDimProps will call dimPropsForPods for all pods in
+// a given a list of pod UIDs
+func (dc *DatapointCache) refreshPodsDimProps(uids []types.UID) {
```

Comment:
Just expand this loop into the functions that call this. I don't like function names that differ by only a single `s` in the middle of them, it is too hard distinguish.

Reply:
You don't even need the intermediate variable `podUIDs` if you inline it.